### PR TITLE
feat(ref): refactor

### PR DESCRIPTION
**Pull request by xakeppok**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,29 +6,20 @@
 
 After your git clone this repository run:
 
-`npm ci --legacy-peer-deps`
+`docker-compose up`
```
Comment (NepipenkoIgor) [inline README.md:9]: What about if i don't want to use docker ?

File: src/brands/brands.controller.ts [hunk 2]
```diff
@@ -15,34 +12,30 @@ export class BrandsController {
...
+		@Query('maxPrice', new ParsePricePipe(true)) maxPrice?: number,
+	): Promise<string[]> {
+		const products: Product[] = await this.productsService.findProducts({
+			subCategory,
+			minPrice,
+			maxPrice,
+		});
+		return products
+			.map((product: Product) => {
+				return product.brand;
```
Comment (NepipenkoIgor) [inline src/brands/brands.controller.ts:37]: Could we do this with mongo but not plain JS?

File: src/main.ts [hunk 3]
```diff
@@ -1,33 +1,35 @@
...
-		.setTitle('NGX-SHOP-API')
-		.setDescription('REST API for angular course')
-		.setVersion('1.0')
-		.build();
-	// tslint:disable-next-line: typedef
-	const document = SwaggerModule.createDocument(app, options);
-	SwaggerModule.setup('', app, document);
+	if (process.env['NODE_ENV'] !== 'production') {
+		const options = new DocumentBuilder()
+			.setTitle('NGX-SHOP-API')
```
Comment (NepipenkoIgor) [inline src/main.ts:23]: What about  production?

File: src/products/interfaces/product-query.interface.ts [hunk 4]
```diff
@@ -1,7 +1,15 @@
 export interface IProductQuery {
-	subCategory: string;
+	subCategory?: string;
+	minPrice?: number;
+	maxPrice?: number;
+	text?: string;
+	brands?: string;
+}
```
Comment (NepipenkoIgor) [inline src/products/interfaces/product-query.interface.ts:7]: Should this to be a separate file ?

File: README.md (hunk 1 above)
Comment (xakeppok) [reply README.md:9]: I will add steps to readme

File: src/products/products.schema.ts [hunk 5]
```diff
@@ -1,49 +1,51 @@
...
-		required: true,
-		type: Number,
-	},
-	subCategory: {
-		required: true,
-		type: Types.ObjectId,
-	},
-});
+// const characteristicsItemsSchema: Schema = new Schema({
+// 	name: { required: true, type: String },
```
Comment (NepipenkoIgor) [inline src/products/products.schema.ts:6]: Who is this ?

File: src/brands/brands.controller.ts (hunk 2 above)
Comment (xakeppok) [reply src/brands/brands.controller.ts:37]: Yes, we can. I will do this.

File: src/main.ts (hunk 3 above)
Comment (xakeppok) [reply src/main.ts:23]: In production we don't need swagger

File: src/products/products.service.ts [hunk 6]
```diff
@@ -39,25 +44,70 @@ export class ProductsService {
...
+			maxPrice,
+			brands,
+		});
+		const productsForPrices: Product[] = await this.findProducts({
+			subCategory,
+			maxPrice: Number.MAX_SAFE_INTEGER,
+			minPrice: 0,
+		});
+		const allPrices: number[] = productsForPrices.map((product: Product): number => product.price);
+		return {
```
Comment (NepipenkoIgor) [inline src/products/products.service.ts:78]: Could we do this with mongo ?

File: src/products/interfaces/product-query.interface.ts (hunk 4 above)
Comment (xakeppok) [reply src/products/interfaces/product-query.interface.ts:7]: I've created separate file for all interfaces, schemas, classes. If you think we dont need this, i can create only 1 file

File: src/products/products.schema.ts (hunk 5 above)
Comment (xakeppok) [reply src/products/products.schema.ts:6]: I need to remove all comments from project

**Pull request merged**